lahja/endpoint.py

```diff
@@ -155,36 +209,69 @@ def has_snappy_support(self) -> bool:
 
         return self._has_snappy_support
 
+    @check_event_loop
     async def start_serving(self, connection_config: ConnectionConfig) -> None:
         """
         Start serving this :class:`~lahja.endpoint.Endpoint` so that it can receive events. Await
         until the :class:`~lahja.endpoint.Endpoint` is ready.
         """
         self._name = connection_config.name
         self._ipc_path = connection_config.path
-        self._create_external_api(self._ipc_path)
-        self._loop = asyncio.get_event_loop()
-        self._internal_loop_running = asyncio.Event(loop=self.event_loop)
-        self._receiving_loop_running = asyncio.Event(loop=self.event_loop)
-        self._internal_queue = asyncio.Queue(loop=self.event_loop)
-        self._receiving_queue = asyncio.Queue(loop=self.event_loop)
-
-        asyncio.ensure_future(self._connect_receiving_queue(), loop=self.event_loop)
-        asyncio.ensure_future(self._connect_internal_queue(), loop=self.event_loop)
+        self._internal_loop_running = asyncio.Event()
+        self._receiving_loop_running = asyncio.Event()
+        self._server_running = asyncio.Event()
+        self._internal_queue = asyncio.Queue()
+        self._receiving_queue = asyncio.Queue()
+
+        # Using `gather` (over e.g. `wait` or plain `ensure_future`) ensures that the inner futures
+        # are automatically cancelled as soon as the parent task is cancelled
+        asyncio.gather(
```

Comment:
Why are we back to using `asyncio.gather` here (was just recently dropped https://github.com/ethereum/lahja/commit/43e479d4159afd99d70de36bac7902a4414cfb0c)

Reply:
Oops, good catch! I accidentally re-introduced it during a rebase.